### ISSUE TITLE
Implement New Game Starter File - Update the following

### ISSUE THREAD

**Issue by flannery-denny** [bootstrapworld/curriculum#1801]
- [x] Save updated versions of all sample games we are providing to teachers - there are many in starter-files.js and linked from the teacher resources page
- [x] Sam the Butterfly
- [x] Player-Animation
- [x] Functions for Character Animation

**Comment by flannery-denny** [comment]
@schanzer Clarifying - The latest game context is 

shared-gdrive("Bootstrap-Algebra-v1.8", "1kupRl2sbC6bb-6G8DAvDkzpiGorljepe")

Confirming that we only want to use this for game files. And not for other algebra files like flags?

**Comment by schanzer** [comment]
@flannery-denny  I can't think of a good reason not to use the most current context with all of our files

**Comment by flannery-denny** [comment]
All algebra files get this context. All DS files get the DS context. What context do CORE starter files get?

**Comment by flannery-denny** [comment]
@schanzer All algebra files get this context. All DS files get the DS context. What context do CORE starter files get?

**Comment by schanzer** [comment]
It depends on what they need! But as far as I know all of them just use the algebra teachpack

**Comment by schanzer** [comment]
@flannery-denny I don't see any context like that in Alice's restaurant. [This is the file I'm looking at](https://code.pyret.org/editor#share=1GScGyigfRPJDdCiGKGnISTXoVBm9f7aE). Where are you looking?

NinjaCat doesn't use a context

**Comment by flannery-denny** [comment]
@schanzer when I open the file you linked and click Choose Context, here's what I see
<img width="921" alt="Screen Shot 2024-02-16 at 9 40 47 AM" src="https://github.com/bootstrapworld/curriculum/assets/66691020/911bf7eb-0ccd-4c94-9291-5e68b0bfc6e4">

I can't open the ninjacat file saved in our googledrive at all (the most recent version I found is saved in [Spring2021](https://drive.google.com/drive/u/0/folders/1lg6CSwGaMP-xN8ujBFKQMNn6V5PtlObf), but I guess you're saying that it doesn't need to be updated so it doesn't matter?

**Comment by schanzer** [comment]
@flannery-denny ohhhh! I didn't realize you were trying to _add_ a context. 

1. Why do you feel this file needs a context? It's never had one before.
2. I think you found a bug in Pyret, where it searches the first line for an existing context and then blindly grabs whatever text comes after the first _n_ characters in `use context `. I'll file a bug report.

And yeah, you can safely ignore ninjacat

**Comment by flannery-denny** [comment]
@schanzer These are the [sample game files](https://drive.google.com/drive/folders/1m1Hsfq2tou-7GMOUXb26ViPuY2aj7Xtl?usp=drive_link) we are offering to teachers. I have updated the contexts, but I think we also need to take a careful look at the code and make sure it is written to be helpful to teachers. Wondering if this job might be more efficient for you to tackle since the updates are fresh in your mind?

**Comment by flannery-denny** [comment]
@schanzer I think that I've successfully brought all of the lessons up to date. The only remaining to do on this issue is make sure we have updated versions of the starter files. I think that I've pulled that off with the exception of the sample game files that you were working on. 

I see these 4 files linked in `starter-files.js`.  Please either finish updating the code and saving both versions in our shared google drive or let me know their status and pass the buck back to me.

**Comment by schanzer** [comment]
@flannery-denny I still have no access to the OG versions of any of these files, so I wasn't able to make any of the changes needed. So here's what I needed to do to put this issue to bed:

1. I made _my own copies_ of all your published files. These copies are now stored in the "OG Files" folder.
2. I went ahead and removed all of your published copies from the "Published Files" folder, and replaced them with published copies of my own. I strongly recommend you delete your published copies from your own Gdrive, to prevent confusion later if you ever search your drive for files with the same name.
3. Then I updated all of the starter file links, confirming that the published copies work the way we want.
4. AFTER all of this was done, I modified the filenames of the OG files so that they all start with "OG". This should make staying organized a lot easier in the future. Note that even after clicking publish again, the filename of the _published_ file didn't change -- so nobody will ever see the letters OG except us.
